relayer/src/keyring/store.rs

```diff
@@ -127,38 +163,108 @@ impl KeyRingOperations for KeyRing {
             account,
         };
 
-        self.insert(key.clone().address, key.clone());
-
         Ok(key)
     }
 
     /// Return a key entry from a key name
-    fn get(&self, address: Vec<u8>) -> Result<&KeyEntry, Error> {
+    fn get_key(&self) -> Result<KeyEntry, Error> {
         match &self {
-            KeyRing::MemoryKeyStore { store: s } => {
-                if !s.contains_key(&address) {
+            KeyRing::MemoryKeyStore {
+                store: s,
+                chain_config: c,
+            } => {
+                if !s.contains_key(c.key_name.clone().as_str()) {
                     Err(Kind::InvalidKey.into())
                 } else {
-                    let key = s.get(&address);
-                    match key {
-                        Some(k) => Ok(k),
+                    let key_content = s.get(c.key_name.as_str());
+                    match key_content {
+                        Some(k) => {
+                            let key_entry = self.key_from_seed_file(k).map_err(|e| {
+                                Kind::KeyStoreOperation.context("failed to get key entry")
+                            })?;
+                            Ok(key_entry)
+                        }
                         None => Err(Kind::InvalidKey.into()),
                     }
                 }
             }
+            KeyRing::TestKeyStore {
+                store: s,
```

Comment:
I would rather use the full names `store` and `chain_config` in the bound variables to improve readability a bit, here and in the other places.

Reply:
Ok, makes sense. I can change that.

Reply:
Pushed changes to fix this https://github.com/informalsystems/ibc-rs/pull/408/commits/f36389685e7ac7a992f74e82bd70f7c7322c0984